suite/tests/client-interface/drx-scattergather.dll.c

```diff
@@ -139,15 +142,28 @@ search_for_next_gather_pc(void *drcontext, instr_t *start_instr, byte **gather_p
         byte *next_pc = decode(drcontext, pc, &temp_instr);
         CHECK(next_pc != NULL,
               "Everything should be decodable in the test until a "
-              "gather instruction will be found.");
+              "scatter or gather instruction will be found.");
         CHECK(!instr_is_cti(&temp_instr), "unexpected cti instruction when decoding");
-        if (instr_is_gather(&temp_instr)) {
-            *gather_pc = pc;
+        if (search_for_gather && instr_is_gather(&temp_instr))
+            break;
+        else if (!search_for_gather && instr_is_scatter(&temp_instr))
             break;
-        }
         pc = next_pc;
     }
     instr_free(drcontext, &temp_instr);
+    return pc;
```

Comment:
This is returning the pc even when it fails to find a gather/scatter.  It should return NULL or sthg if it fails.

Reply:
If there is no scatter/gather, this function would already be in a indefinite loop or would fail the CHECK. It relies on the test's code and the fact that there is always a scatter/gather.

Reply:
IMHO this is not clean code: this function looks broken.   It should have an assert that it found sthg, or *some* indication that it's not a mistake that nobody noticed.

Reply:
Fair enough, done!